unidist/cli/__main__.py

```diff
@@ -0,0 +1,119 @@
+# Copyright (C) 2021 Modin authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""Command line interface for unidist."""
+
+import os
+import sys
+import argparse
+
+try:
+    import unidist  # noqa: F401
+except ImportError:
+    sys.path.insert(
+        0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../.."))
+    )
+
+from unidist.cli.base.utils import Defaults
+from unidist.core.base.common import BackendName
+
+
+def main():
+    """Run an application with unidist."""
+    usage_examples = [
+        "\n\tIn case 'unidist' is installed as a python package, run binary:",
+        "\n\tunidist script.py  # Ray backend is used by default",
+        f"\n\tunidist script.py --backend {BackendName.MPI}  # MPI backend is used",
+        f"\n\tunidist -m pytest script.py -b {BackendName.DASK}  # Dask backend is used, running the script using 'pytest'",
+        f"\n\tunidist script.py -b {BackendName.MP} --num_cpus 16 # MultiProcessing backend is used and uses 16 CPUs",
+        "\n\n\tTo run from sources run 'unidist/cli':",
+        f"\n\tpython unidist/cli script.py -b {BackendName.MPI} --num_cpus 16 -hosts localhost  # MPI backend uses 16 workers on 'localhost' node",
+        f"\n\tpython unidist/cli script.py -b {BackendName.MPI} -num_cpus 2 4 --hosts localhost x.x.x.x  # MPI backend uses 2 workers on 'localhost' and 4 on 'x.x.x.x'",
+    ]
+    parser = argparse.ArgumentParser(
+        description="Run python code with 'unidist'.",
+        usage="".join(usage_examples),
+    )
+
+    required_args_group = parser.add_argument_group("required arguments")
+    ray_specific_args_group = parser.add_argument_group(
+        "Ray backend-specific arguments"
+    )
+    required_args_group.add_argument("script", help="specify a script to be run")
+    parser.add_argument(
+        "-b",
+        "--backend",
+        type=str,
+        choices=[
+            BackendName.RAY,
+            BackendName.MPI,
+            BackendName.DASK,
+            BackendName.MP,
+            BackendName.PY,
+        ],
+        default=Defaults.BACKEND,
+        help=f"specify an execution backend. Default value is taken from 'UNIDIST_BACKEND' environment variable. If 'UNIDIST_BACKEND' isn't set, '{BackendName.RAY}' backend is used",
+    )
+    parser.add_argument(
+        "-m",
```

Comment:
Why don't we specify a type for this parameter?

Reply:
`str` is used by default, explicit `type` was added